crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -922,5 +1084,114 @@ macro_rules! periodic_job {
         }
     };
 }
+
 pub(crate) use periodic_job;
 pub(crate) use try_and_log;
+
+#[cfg(test)]
+mod tests {
+    use floresta_chain::pruned_utreexo::partial_chain::PartialChainState;
+
+    use crate::node::UtreexoNode;
+    use crate::running_node::RunningNode;
+
+    #[test]
+    fn test_parse_address() {
+        let ipv6_valid = "[::1]";
+        let ipv6_invalid = "[::1";
+        let ipv6_with_port = "[::1]:8333";
+        let ipv6_with_invalid_port = "[::1]:8333:8333";
+
+        let ipv4_valid = "127.0.0.1";
+        let ipv4_invalid = "321.321.321.321";
+        let ipv4_with_port = "127.0.0.1:8333";
+        let ipv4_with_invalid_port = "127.0.0.1:8333:8333";
+
+        let hostname_valid = "example.com";
+        let hostname_invalid = "example";
+        let hostname_with_port = "example.com:8333";
+        let hostname_with_invalid_port = "example.com:8333:8333";
+
+        UtreexoNode::<RunningNode, PartialChainState>::resolve_connect_host(ipv6_valid, 8333)
+            .unwrap();
+
+        assert!(
```

Comment:
Maybe use `assert!(..., "message")` to provide better debugging information when a test fails. This helps identify which specific case failed without needing to look into the inputs.

```rust
assert!(
    UtreexoNode::<RunningNode, PartialChainState>::resolve_connect_host(ipv6_invalid, 8333)
        .is_err(),
    "Expected error for invalid IPv6 address: {ipv6_invalid}"
);

```